src/NServiceBus.Core/Extensibility/OutgoingPipelineExtensionState.cs

```diff
@@ -0,0 +1,30 @@
+namespace NServiceBus.Extensibility
+{
+    using System;
+
+    /// <summary>
+    /// Marks a given piece of state as an extension that needs to be handled before the message body is serialized.
+    /// </summary>
+    public abstract class OutgoingPipelineExtensionState
```

Comment:
The idea here is that all the stare object added as extensions should be handled by corresponding behaviors. There is nothing preventing you from using the extension methods ob the `SendOptions` with a corresponding feature disabled. This thing tries to address that flow by throwing an exception rather than ignoring.